### PR TITLE
Publish android artifacts to s3

### PR DESCRIPTION
**Pull request by oguzkocer**
## Description

This PR updates `react-native-bridge` and `react-native-aztec` libraries to publish their artifacts to S3 instead of Bintray. This is due to [Bintray shutting down their services](https://jfrog.com/blog/into-the-sunset-bintray-jcenter-gocenter-and-chartcenter/).

## How has this been tested?

Gutenberg-mobile CI has a `Build Android RN Bridge & Publish to S3` task which uses these changes to upload the artifacts to S3 which was successful in https://github.com/wordpress-mobile/gutenberg-mobile/pull/3325. The uploaded artifacts are used in https://github.com/wordpress-mobile/WordPress-Android/pull/14385 for WPAndroid which should be tested to verify these changes.

## Types of changes

Build file changes for `react-native-bridge` and `react-native-aztec` libraries.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/react-native-aztec/android/build.gradle
Comment (jkmassel) [inline packages/react-native-aztec/android/build.gradle:126]: Nit: can we reference the `group` variable declared above for this, or do they have to be separate?

Same comment for the other `build.gradle` file

**Review by jkmassel — APPROVED** [review]
This looks very much the same as the other configuration, so I'm confident it will work as well.

Approved pending adding any required configuration stuff to CircleCI/GitHub Actions (not sure which this uses right now – this doesn't seem to be called from CI just yet)

Also left one note for your consideration

File: packages/react-native-aztec/android/build.gradle
Comment (oguzkocer) [reply packages/react-native-aztec/android/build.gradle:126]: Good point - but let me address that as a follow up PR just so we can merge this one as soon as possible.

**Comment by jd-alexander** [comment]
Hi @oguzkocer I noticed that we still have three bintray maven repositories in the codebase. References below:

https://github.com/WordPress/gutenberg/blob/afa2d8cc9b239ca4b677ef7838d6cf43db7b84ba/packages/react-native-aztec/android/build.gradle#L88

https://github.com/WordPress/gutenberg/blob/afa2d8cc9b239ca4b677ef7838d6cf43db7b84ba/packages/react-native-bridge/android/build.gradle#L19

https://github.com/WordPress/gutenberg/blob/afa2d8cc9b239ca4b677ef7838d6cf43db7b84ba/packages/react-native-bridge/android/react-native-bridge/build.gradle#L66

Should these be removed?

**Comment by oguzkocer** [comment]
@jd-alexander `react-native-mirror` ones are definitely necessary, but I am not 100% sure if the `maven` one is. They'll be removed soon as part of another PR though, so I think it's safe to ignore them in this PR.

**Review by jd-alexander — APPROVED** [review]
The binary build publishing with S3 works as expected. Verified with WP Android https://github.com/wordpress-mobile/WordPress-Android/pull/14385#pullrequestreview-641450516 LGTM 🚢 This PR will need to be updated with `trunk` before merging.

**Comment by guarani** [comment]
> The binary build publishing with S3 works as expected. Verified with WP Android [wordpress-mobile/WordPress-Android#14385 (review)](https://github.com/wordpress-mobile/WordPress-Android/pull/14385#pullrequestreview-641450516) LGTM 🚢 This PR will need to be updated with `trunk` before merging.

👋 Could you please clarify why updating with `trunk` is needed?

**Comment by guarani** [comment]
I'd be happy to update with `trunk`, just want to make sure I understand why since it'll re-run the tests all over again.

**Comment by jd-alexander** [comment]
Hi @guarani I mentioned this due to the gb-mobile conflict. So I assumed I would update with `trunk` to get that removed. Is that it doesn't matter when I update the gb-mobile PR with a merge commit? Just checking to ensure I have the right idea.
Screenshot 2021-04-23 at 9 56 52 AM

**Comment by guarani** [comment]
> Hi @guarani I mentioned this due to the gb-mobile conflict. So I assumed I would update with `trunk` to get that removed. Is that it doesn't matter when I update the gb-mobile PR with a merge commit? Just checking to ensure I have the right idea.

Oh I see, I hadn't noticed the conflict in Gutenberg Mobile. 
My feeling is that we could merge this anyway as-is, since it'll be merged on to the top of `trunk` here. Then we'll update the Gutenberg Mobile PR to point to the merge commit and then merge that. wdyt?

**Comment by jd-alexander** [comment]
> Oh I see, I hadn't noticed the conflict in Gutenberg Mobile.
My feeling is that we could merge this anyway as-is, since it'll be merged on to the top of trunk here. Then we'll update the Gutenberg Mobile PR to point to the merge commit and then merge that. wdyt?

sounds good! 🙇🏾

**Pull request merged**
